packages/ui/src/Select.tsx

```diff
@@ -0,0 +1,97 @@
+import { DataTestProp } from '@hazelcast/helpers'
+import React, { FC, SelectHTMLAttributes, useRef, FocusEvent, ChangeEvent } from 'react'
+import { v4 as uuid } from 'uuid'
+import cn from 'classnames'
+import { ChevronDown } from 'react-feather'
+
+import { Error, errorId } from './Error'
+import { Label } from './Label'
+import { Icon } from './Icon'
+
+import styles from './Select.module.scss'
+
+export type SelectOption = Pick<HTMLOptionElement, 'value' | 'text'> & {
+  disabled?: boolean
+}
+
+export type SelectCoreProps = {
+  options: SelectOption[]
+  name: string
+  value?: string
+  onBlur?: (e: FocusEvent<HTMLSelectElement>) => void
+  onChange?: (e: ChangeEvent<HTMLSelectElement>) => void
+  error?: string
+}
+export type SelectExtraProps = {
+  label: string
+  className?: string
+  selectClassName?: string
+  errorClassName?: string
+  children?: never
+  notSelectedPlaceholder?: string
+} & DataTestProp &
+  Pick<SelectHTMLAttributes<HTMLSelectElement>, 'autoFocus' | 'disabled' | 'required'>
+
+export type SelectProps = SelectExtraProps & SelectCoreProps
+
+export const Select: FC<SelectProps> = ({
+  autoFocus,
+  'data-test': dataTest,
+  className,
+  disabled,
+  error,
+  errorClassName,
+  label,
+  name,
+  onBlur,
+  onChange,
+  options,
+  required,
+  selectClassName,
+  value,
+  notSelectedPlaceholder = 'Select',
+}) => {
+  const idRef = useRef(uuid())
+
+  return (
+    <div
+      data-test={dataTest}
+      className={cn(
+        styles.container,
+        {
+          [styles.disabled]: disabled,
+          [styles.hasError]: error,
+          [styles.empty]: !value,
+        },
+        className,
+      )}
+    >
+      <Label id={idRef.current} label={label} />
+      <div className={styles.selectContainer}>
+        <select
```

Comment:
If we use native `select` than our `Select` is going to look quite differently compared to `Multiselect`. Should we use some library that can cover both cases for consistent look and feel?

Reply:
I think `react-select` worked out just perfectly for us at MC. I am researching other options now.

Reply:
@aigoncharov I agree with using one lib for both scenarios

Reply:
Good point! I totally overlooked the fact there will be a multi-select component. As much as I don't like `react-select` styling API, I think we should stick to it. It has proven to be a great match for us.

Reply:
@aigoncharov @jankoritak we already use react-select as well on .cloud.

Reply:
@jankoritak @jankoritak there is an alternative to `react-select`. We could use low-level [downshift](https://github.com/downshift-js/downshift) and build a custom select ourselves. We already have `popper` to position the menu, so we'll just need to style it. However, I could imagine it might take a while to do it right.

Reply:
I gave it some thinking and I feel like we would be better off with `reatc-select`. It might be a bit bulky, but we have saved KB with other components, right? Time is of the matter, we should start integration ASAP. We could always reimplement the component matching the existing API later on when we have time. Let's do simple stuff for now. WDYT?

Reply:
@jankoritak could you prioritize this task over your other ones? I think `Select` and `Password` are only ones left to start with the new login flow for MC.

Reply:
Certainly, I'll start with this right away!

Reply:
Implementing the components(s) using downshift would take a lot of time. I would stick to `react-select`, at least for now.